src/test/java/org/worshipsongs/importer/SongParserTest.java

```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty("omit-xml-declaration", "yes");
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+            System.out.println(out.toString());
+
+            assertEquals("<verse type=\"\" label=\"\"><![CDATA[data]]></verse>", out.toString());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testGetLyricsTag()
+    {
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+
+            Element lyricsTag = parser.getLyricsTag(document);
+            document.appendChild(lyricsTag);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty("omit-xml-declaration", "yes");
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+
+            assertEquals("<lyrics/>", out.toString());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testGetSongTag()
+    {
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+            document.setXmlStandalone(true);
+
+            Element songTag = parser.getSongTag(document);
+            document.appendChild(songTag);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+            System.out.println(out.toString());
+            assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\"?><song version=\"1.0\"/>", out.toString());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testParseVerseOrders()
+    {
+        assertEquals("v1", parser.parseVerseOrders("v1 v2 v3 v4")[0]);
+        assertEquals("v2", parser.parseVerseOrders("v1 v2 v3 v4")[1]);
+        assertEquals("v3", parser.parseVerseOrders("v1 v2 v3 v4")[2]);
+        assertEquals("v4", parser.parseVerseOrders("v1 v2 v3 v4")[3]);
+    }
+
+    @Test
+    public void testParseVerseType()
+    {
+        assertEquals("v", parser.parseVerseType("v1"));
+        assertEquals("c", parser.parseVerseType("c1"));
+    }
+
+    @Test
+    public void testParseVerseLabel()
+    {
+        assertEquals("1", parser.parseVerseType("v1"));
+        assertEquals("2", parser.parseVerseType("v2"));
```

Comment:
Change this test to parse verse type from the text "V2". Please note that "V" is upper case here.